File: python/nt_solutions/ml_3_classification/exo7.py
niter = 1000
W = np.zeros((p,k))
Elist = np.zeros((niter,1))
tau = .01
for i in np.arange(0,niter):
    W = W - tau * nablaE(W)
    Elist[i] = E(W)
## display ##
plt.clf
plt.plot(np.arange(0,niter), Elist)
plt.axis('tight')

File: r/nt_solutions/ml_3_classification/exo7.R
W = matrix(0, nrow=p, ncol=k)
Elist = c()
tau = 0.01
niter = 1000
for (i in 1:niter)
{
    W = W - tau * nablaE(W)
    Elist = c(Elist, E(W))
}

plot(c(1: niter), Elist, type="l" ,col="blue", xlab="Iteration")